docs/technical_reference/tools/parallel_manager.rst

```diff
@@ -0,0 +1,105 @@
+Parallel Manager
+====================
+
+.. index::
+    pair: watertap.tools.parallel.parallel_manager;parallel_manager
+
+.. currentmodule:: watertap.tools.parallel.parallel_manager
+
+Overview
+--------
+
+The parallel manager is a high level wrapper designed for the parameter sweep
+tool that allows the use of different parallel paradigms for running
+parameter sweep in parallel, for example, on a supercomputer, using a unified
+API. Currently, supported parallel backends include:
+
+    #. MPI
+    #. Multiprocessing
+    #. Concurrent Futures
+    #. Ray Core
+
+Parallel manager was implemented to support different workflows of WaterTAP
+users and developers while reducing the burden to develop software. In 
+particular, it supports collective operations using the various paradigms.
+These operations are listed below:
+
+    #. Barrier : Waits until all parallel workers arrive at the same point in code
+    #. Gather : Gathers an array or list on one or all parallel workers.
+    #. Scatter : Divides up the array or list across all the parallel workers
+    #. Broadcast : Sends the array or list to all parallel workers
+    #. Reduce : Collects the data from all parallel worker and performs a mathematical operation. The current behavior sums the values from all workers.
+
+Usage
+--------
+
+End User
+~~~~~~~~
+
+If the user is simply invoking the parameter sweep tool, they only need to
+specify parallel backend by using the keyword argument ``parallel_back_end``
+when constructing a parameter sweep object. Valid keyword values are
+
+    #. "MPI"
+    #. "ConcurrentFutures"
+    #. "MultiProcessing"
+    #. "RayIo"
```

Comment:
I'm wondering if there is a very brief description that can be provided for each. Why would I choose, say, `RayIo `over  `ConcurrentFutures`? Perhaps it would be hard to concisely nail down pros/cons of going with any of these options. If that's the case, can we point the user to some reference that might inform decisions on which capabilities to try? 

Additionally, can we provide some **general** recommendations or examples to offer guidance on getting started, e.g., "if using a personal laptop, users could make use of "MultiProcessing""

Reply:
I just added more information for when to use which parallel backend.